python/device/device.py

```diff
@@ -380,14 +429,19 @@ def on_pingback_received(pingback_id):
                 self.metrics.d2c_received_by_service_app.increment()
 
             with self.pingback_list_lock:
-                self.pingback_wait_list[pingback_id] = (on_pingback_received, (pingback_id,))
+                self.pingback_wait_list[pingback_id] = MessageWaitingToArrive(
+                    callback=on_pingback_received,
+                    pingback_id=pingback_id,
+                    message=msg,
+                    send_time=time.time(),
```

Comment:
is this utc time?

Reply:
unix epoch time stored as an integer.  I think I'll call this "timestamp" instead of "time," unless you have a better way to distinguish between a Time object and a unix epoch time integer.

---
In reply to: [512845578](https://github.com/Azure/iot-sdk-longhaul/pull/17#discussion_r512845578) [](ancestors = 512845578)

Reply:
Actually, it's probably a float instead of an integer.

---
In reply to: [512874594](https://github.com/Azure/iot-sdk-longhaul/pull/17#discussion_r512874594) [](ancestors = 512874594,512845578)

Reply:
But I'll change the name in a future commit since it will lead to merge conflicts for me right now.

---
In reply to: [512874969](https://github.com/Azure/iot-sdk-longhaul/pull/17#discussion_r512874969) [](ancestors = 512874969,512874594,512845578)